primitives/state-machine/src/overlayed_changes/changeset.rs

```diff
@@ -62,32 +63,46 @@ pub enum ExecutionMode {
 
 #[derive(Debug, Default, Clone)]
 #[cfg_attr(test, derive(PartialEq))]
-struct InnerValue {
+pub struct InnerValue<V> {
```

Comment:
Why does this need to be `pub` now? We should try to keep encapsulation alive.

Reply:
I will look into it, just did realize that basti is also working on the issue so maybe there will be other changes incomming.

Reply:
Indeed seems like an error from me.